pkg/nfd-gc/nfd-gc.go

```diff
@@ -115,17 +115,19 @@ func (n *nfdGarbageCollector) deleteNodeHandler(object interface{}) {
 		obj = deletedFinalStateUnknown.Obj
 	}
 
-	node, ok := obj.(*corev1.Node)
+	meta, ok := obj.(*metav1.PartialObjectMetadata)
 	if !ok {
-		klog.InfoS("cannot convert object to v1.Node", "object", object)
+		klog.InfoS("cannot convert object to metav1.ObjectMeta", "object", object)
 		return
 	}
+	nodeName := meta.ObjectMeta.GetName()
 
-	n.deleteNRT(node.GetName())
+	n.deleteNRT(nodeName)
 
 	// Delete all NodeFeature objects (from all namespaces) targeting the deleted node
-	nfListOptions := metav1.ListOptions{LabelSelector: nfdv1alpha1.NodeFeatureObjNodeNameLabel + "=" + node.GetName()}
-	if nfs, err := n.nfdClient.NfdV1alpha1().NodeFeatures("").List(context.TODO(), nfListOptions); err != nil {
+	gvr := nfdv1alpha1.SchemeGroupVersion.WithResource("nodefeatures")
```

Comment:
make a module var out of it?
```go
var (
   nodeFeatureGvr = nfdv1alpha1.SchemeGroupVersion.WithResource("nodefeatures")
   nrtGvr = topologyv1alpha2.SchemeGroupVersion.WithResource("noderesourcetopologies")
)
```

Reply:
Good point. Addressed in the "base PR" in #1813

Reply:
Fixed